File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/TrackerContract.java
package ua.com.foxminded.locationtrackera.ui.tracker;

import ua.com.foxminded.locationtrackera.mvi.FragmentContract;
import ua.com.foxminded.locationtrackera.ui.tracker.state.TrackerScreenEffect;
import ua.com.foxminded.locationtrackera.ui.tracker.state.TrackerScreenState;

public class TrackerContract {

    public interface ViewModel extends FragmentContract.ViewModel<TrackerScreenState, TrackerScreenEffect> {
        void logout();

        void setDialogResponse(int code);

        void setSharedPreferencesServiceFlag(boolean flag);
    }

    public interface View extends FragmentContract.View {
        void proceedToSplashScreen();

        void changeGpsStatus(int gpsStatus);

        void changeServiceStatus(boolean isEnabled);

        void showDialogFragment(int argType, int message, int negativeButton, int positiveButton);
    }

    public interface Host extends FragmentContract.Host {
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/TrackerContract.kt
package ua.com.foxminded.locationtrackera.ui.tracker

import ua.com.foxminded.locationtrackera.ui.tracker.state.TrackerScreenEffect
import ua.com.foxminded.locationtrackera.ui.tracker.state.TrackerScreenState
import ua.com.foxminded.locationtrackera.mvi.fragments.FragmentContract

class TrackerContract {
    interface ViewModel : FragmentContract.ViewModel<
            TrackerScreenState,
            TrackerScreenEffect> {
        fun logout()
        fun setDialogResponse(code: Int)
        fun setSharedPreferencesServiceFlag(flag: Boolean)
    }

    interface View : FragmentContract.View {
        fun proceedToSplashScreen()
        fun changeGpsStatus(gpsStatus: Int)
        fun changeServiceStatus(isEnabled: Boolean)
        fun showDialogFragment(argType: Int, message: Int, negativeButton: Int, positiveButton: Int)
    }

    interface Host : FragmentContract.Host
}